Apps/Admin/Tests/Services/BetaFeatureServiceTests.cs

```diff
@@ -0,0 +1,346 @@
+            [
+                GenerateBetaFeatureCode(BetaFeature.Salesforce),
+            ];
+
+            IList<UserProfile> userProfiles =
+            [
+                GenerateUserProfile(Hdid1Email1, Email1, betaFeatureCodes),
+                GenerateUserProfile(Hdid2Email1, Email1, betaFeatureCodes),
+            ];
+
+            IList<BetaFeatureAccess> betaFeatureAssociations =
+            [
+                GenerateBetaFeatureAccess(Hdid1Email1, Email1),
+                GenerateBetaFeatureAccess(Hdid2Email1, Email1),
+                GenerateBetaFeatureAccess(Hdid3Email2, Email2),
+            ];
+
+            Mock<IUserProfileDelegate> profileDelegateMock = new();
+            profileDelegateMock.Setup(s => s.GetUserProfilesAsync(Email1, true, It.IsAny<CancellationToken>())).ReturnsAsync(userProfiles);
+
+            Mock<IBetaFeatureAccessDelegate> betaFeatureAccessDelegateMock = new();
+            betaFeatureAccessDelegateMock.Setup(s => s.GetAllAsync(true, It.IsAny<CancellationToken>())).ReturnsAsync(betaFeatureAssociations);
+            IBetaFeatureService service = GetBetaFeatureService(betaFeatureAccessDelegateMock: betaFeatureAccessDelegateMock);
+
+            int expectedCount = userProfiles.Count;
+
+            // Act
+            IEnumerable<Common.Models.BetaFeatureAccess> enumerable = await service.GetBetaFeatureAccessAsync();
+            IList<Common.Models.BetaFeatureAccess> actual = enumerable.ToList();
+
+            // Assert
+            Assert.Equal(expectedCount, actual.Count);
+            Assert.Equal(Email1, actual.ElementAt(0).Email);
+            Assert.Equal(Email2, actual.ElementAt(1).Email);
+        }
+
+        /// <summary>
+        /// SetUserAccessAsync.
+        /// </summary>
+        /// <param name="existingBetaFeature">User profile's current associated beta feature.</param>
+        /// <param name="betaFeature">Beta feature to associate to user profile.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous unit test.
+        /// </returns>
+        [InlineData(BetaFeature.Salesforce, Common.Constants.BetaFeature.Salesforce)]
+        [InlineData(null, Common.Constants.BetaFeature.Salesforce)]
+        [InlineData(BetaFeature.Salesforce, null)]
+        [InlineData(null, null)]
+        [Theory]
+        public async Task ShouldSetUserAccess(BetaFeature? existingBetaFeature, Common.Constants.BetaFeature? betaFeature)
+        {
+            // Arrange
+            IList<Common.Constants.BetaFeature> betaFeatures = betaFeature != null
+                ?
+                [
+                    betaFeature.Value,
+                ]
+                : [];
+
+            ICollection<BetaFeatureCode> betaFeatureCodes = existingBetaFeature != null
+                ?
+                [
+                    GenerateBetaFeatureCode(existingBetaFeature.Value),
+                ]
+                : [];
```

Comment:
Nit: formatting for readability.

```suggestion
            IList<Common.Constants.BetaFeature> betaFeatures = betaFeature != null
                ? [betaFeature.Value]
                : [];

            ICollection<BetaFeatureCode> betaFeatureCodes = existingBetaFeature != null
                ? [GenerateBetaFeatureCode(existingBetaFeature.Value)]
                : [];
```